### PR TITLE
Apply new keyword arguments syntax

### PR DESCRIPTION
**Pull request by K-S-A**
This PR adds ruby 3.0 compatibility.

Fixes: https://github.com/brandonhilkert/sucker_punch/issues/234

### CODE REVIEW

**Comment by K-S-A** [comment]
Probably, it will be better to add backward compatibility with ruby 2.x versions.

**Comment by brandonhilkert** [comment]
For sure. Would the current be compatible?

File: lib/sucker_punch/job.rb [hunk 1]
```diff
@@ -32,17 +32,17 @@ def logger
     end
 
     module ClassMethods
-      def perform_async(*args)
+      def perform_async(*args, **kwargs)
         return unless SuckerPunch::RUNNING.true?
         queue = SuckerPunch::Queue.find_or_create(self.to_s, num_workers, num_jobs_max)
-        queue.post(args) { |job_args| __run_perform(*job_args) }
+        queue.post([args, kwargs]) { |job_args, job_kwargs| __run_perform(*job_args, **job_kwargs) }
```
Comment (K-S-A) [inline lib/sucker_punch/job.rb:38]:
```ruby
queue.post { __run_perform(*args, **kwargs) }
```

I think, that this version will work in the same way. I'll need to double-check implementation in concurrent-ruby.

**Comment by K-S-A** [comment]
> For sure. Would the current be compatible?

Not yet. I'll need to make few tweaks. E.g. replace recently introduced "...".

**Comment by K-S-A** [comment]
@brandonhilkert , is it necessary to support ruby versions before ruby 2.5?

**Comment by brandonhilkert** [comment]
Yes, I'd like to. What changes would have to be excluded? Although it could be a major bump, but I always find that to be extremely frustrating as a user.

**Comment by K-S-A** [comment]
I'll check PR tomorrow morning my time and will add final fixes and some test cases, if necessary.

File: lib/sucker_punch/job.rb (hunk 1 above)
Comment (K-S-A) [reply lib/sucker_punch/job.rb]: I've tracked down argument forwarding to this place: https://github.com/ruby-concurrency/concurrent-ruby/blob/master/lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb#L351

Both `args` and `job_args` are references to the same object.

File: lib/sucker_punch/job.rb [hunk 2]
```diff
@@ -55,14 +55,20 @@ def max_jobs(num)
         self.num_jobs_max = num
       end
 
-      def __run_perform(*args)
+      def __run_perform(*args, **kwargs)
         SuckerPunch::Counter::Busy.new(self.to_s).increment
-        result = self.new.perform(*args)
+
+        result = if RUBY_VERSION >= '2.5'
```
Comment (K-S-A) [inline lib/sucker_punch/job.rb:61]: Checked versions: 2.4.0, 2.5.0, 2.6.0, 2.7.2, and 3.0.0. Tests are green.

**Comment by brandonhilkert** [comment]
Could you add an entry to the change log?

**Comment by K-S-A** [comment]
@brandonhilkert , I've updated `CHANGES.md`. Not sure about the exact version number, though.

**Comment by brandonhilkert** [comment]
Let's go with 3.0.0. Seems fitting ;)

**Comment by K-S-A** [comment]
I do like 3.0.0)
Done.

**Comment by brandonhilkert** [comment]
`3.0.0` has been released. Thanks again!

**Comment by eregon** [comment]
`(*args, **kwargs)`-delegation is unfortunately incorrect on Ruby <= 2.7.
In other words, it's only correct in Ruby 3+.
See https://eregon.me/blog/2021/02/13/correct-delegation-in-ruby-2-27-3.html for correct delegation compatible with Ruby 2 & 3.

**Comment by ivoanjo** [comment]
For context, it also triggered an error in JRuby for our test suite @ https://github.com/DataDog/dd-trace-rb/issues/1369 which is why I ended up reporting it TruffleRuby && JRuby.

**Comment by brandonhilkert** [comment]
Fixed in `3.0.1`.

**Comment by eregon** [comment]
Thanks for the quick fix and release :) (https://github.com/brandonhilkert/sucker_punch/pull/236)

**Pull request merged**